iree/compiler/Codegen/Utils/Utils.cpp

```diff
@@ -140,6 +140,7 @@ SmallVector<int64_t> getUntiledResultShape(linalg::LinalgOp linalgOp,
 
   for (auto pair : llvm::zip(llvm::reverse(partitionedLoops), loopInfo)) {
     unsigned loopIndex = std::get<0>(pair);
+    if (loopIndex >= outputShape.size()) continue;
```

Comment:
In what case does this happen? I would think the output rank matches the iteration space? Maybe I'm not missing something about this logic.

Reply:
Not necessary. For example, for `linalg.nhwc`, we have `(N, OH, OW, KH, KW, C)` but it's output only have `(N, OH, OW, C)`. If we just use the `loopIndex` for `C` (which is 5) to access the output shape, it would be OOB.

But on a second thought, this is actually not the proper fix. The real issue is that it's assuming the first iterators of the Linalg op are always used to access output shape dimensions. That's not true; instead we need to figure out which loops are used to access output shape dimensions. I've fixed this in a second commit. Thanks for spotting this!

Reply:
In this case KH and KW would be reduction dimensions and wouldn't be part of `partitionedLoops`, right? I would expect all the loops in partiotionedLoops to be parallel loop and match to an output dimension but I could be missing something.

Reply:
Yes, `KW` and `KH` won't be partitioned (so not in `partitionedLoops`) but `C` can be in it. The `loopIndex` for `C` is 5, which is greater than the output shape's rank. Accessing `outputShape` with `loopIndex` can see OOB for such case.

Reply:
For pooling, we normally want to distribute along `OH`, `OW`, and `C`. `N` is typically one; no need to distribute while `C` can be quite large.